sim/src/node.rs

```diff
@@ -61,16 +61,29 @@ impl SimElement for Node {
             self.queue.size() + (self.generation_rate as usize),
             self.egress_rate as usize,
         ) {
-            // send the rpc to a random neighbor, if there are any
+            // send the rpc to a random neighbor, if no neighbor specified
             let rpc: Rpc;
             if self.queue.size() > 0 {
                 let deq = self.dequeue(tick);
                 rpc = deq.unwrap();
             } else {
-                rpc = Rpc::new_rpc(tick as u32);
+                rpc = Rpc::new_rpc(&tick.to_string());
             }
             let neigh_len = self.neighbors.len();
-            if neigh_len > 0 {
+            if rpc.headers.contains_key("dest") {
+                let mut have_dest = false;
+                let dest = rpc.headers["dest"].clone();
```

Comment:
What is going on with the clone and then referencing here? This probably can be cleaned up.  dest only needs to be cloned at the end when we store it in the rpc?